test/framework/alertmanager.go

```diff
@@ -231,24 +249,61 @@ func (f *Framework) WaitForAlertmanagerReady(ctx context.Context, ns string, a *
 	return nil
 }
 
-func (f *Framework) UpdateAlertmanagerAndWaitUntilReady(ctx context.Context, ns string, a *monitoringv1.Alertmanager) (*monitoringv1.Alertmanager, error) {
-	a, err := f.MonClientV1.Alertmanagers(ns).Update(ctx, a, metav1.UpdateOptions{})
+func (f *Framework) PatchAlertmanagerAndWaitUntilReady(ctx context.Context, name, ns string, spec monitoringv1.AlertmanagerSpec) (*monitoringv1.Alertmanager, error) {
+	a, err := f.PatchAlertmanager(ctx, name, ns, spec)
 	if err != nil {
-		return nil, err
+		return nil, errors.Wrapf(err, "failed to patch Alertmanager %s/%s", ns, name)
+	}
+
+	err = f.WaitForAlertmanagerReady(ctx, a)
+	if err != nil {
+		return nil, fmt.Errorf("failed to update Alertmanager: %v", err)
 	}
 
-	err = f.WaitForPodsReady(
+	return a, nil
+}
+
+func (f *Framework) PatchAlertmanager(ctx context.Context, name, ns string, spec monitoringv1.AlertmanagerSpec) (*monitoringv1.Alertmanager, error) {
+	b, err := json.Marshal(
+		&monitoringv1.Alertmanager{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       monitoringv1.AlertmanagersKind,
+				APIVersion: schema.GroupVersion{Group: monitoring.GroupName, Version: monitoringv1.Version}.String(),
+			},
+			Spec: spec,
+		},
+	)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to marshal Alertmanager spec")
+	}
+
+	p, err := f.MonClientV1.Alertmanagers(ns).Patch(
 		ctx,
-		ns,
-		5*time.Minute,
-		int(*a.Spec.Replicas),
-		alertmanager.ListOptions(a.Name),
+		name,
+		types.ApplyPatchType,
+		b,
+		metav1.PatchOptions{
+			Force:        func(b bool) *bool { return &b }(true),
```

Comment:
Can't we use `"k8s.io/utils/pointer"` here?
```suggestion
			Force:        pointer.Bool(true),
```